### PR TITLE
Support the format returned by StringFromGUID2 in guid.parse.

### PR DESCRIPTION
**Pull request by jlaanstra**
Support the format returned by StringFromGUID2 in guid.parse. ```{(})```  are common characters used in guid formats and it would be convenient if cppwinrt would automatically handle them.

### CODE REVIEW

File: strings/base_types.h [hunk 1]
```diff
@@ -85,8 +85,11 @@ WINRT_EXPORT namespace winrt
     private:
 
         template <typename TStringView>
-        static constexpr guid parse(TStringView const value)
```
Comment (sylveon) [inline strings/base_types.h:88]: Why remove `const` here?

Comment (sylveon) [reply strings/base_types.h:88]: Oh, I missed that `value.remove_prefix` was called right under

File: strings/base_types.h [hunk 2]
```diff
@@ -85,8 +85,11 @@ WINRT_EXPORT namespace winrt
...
+        static constexpr guid parse(TStringView value)
         {
+            value.remove_prefix(std::min(value.find_first_not_of(" {("), v.size()));
```
Comment (sylveon) [inline strings/base_types.h:90]: I don't think this compiles when TStringView is std::wstring_view

File: strings/base_types.h (hunk 1 above)
Comment (jlaanstra) [reply strings/base_types.h:88]: Can't modify a const value.

File: strings/base_types.h (hunk 2 above)
Comment (jlaanstra) [reply strings/base_types.h]: You're right. Went with the simpler option.

File: strings/base_types.h [hunk 3]
```diff
@@ -87,8 +87,12 @@ WINRT_EXPORT namespace winrt
         template <typename TStringView>
         static constexpr guid parse(TStringView value)
         {
-            value.remove_prefix(std::min(value.find_first_not_of(" {("), v.size()));
-            value.remove_suffix(std::min(value.find_last_not_of(" })"), v.size()));
+            // Handle {} and ()
+            if (value.size() == 38 && ((value[0] == '{' && value[37] == '}') || (value[0] == '(' && value[37] == ')')))
```
Comment (kennykerr) [inline strings/base_types.h:91]: Cool! I know `{guid}` is common but I'm not sure I've seen `(guid)` very often... What's the motivation behind that?

Comment (jlaanstra) [reply strings/base_types.h:91]: Based on what the .NET version of Guid.Parse supports: https://docs.microsoft.com/en-us/dotnet/api/system.guid.parse?view=net-6.0

**Review by kennykerr — APPROVED** [review]

**Pull request merged**